### ISSUE TITLE
restrict-plus-operands false positive with ternary operator

### ISSUE THREAD

**Issue by buu700** [palantir/tslint#1922]
### Bug Report

- __TSLint version__: `4.2.0`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const balls = 'hello' + (someBoolean ? 'a' : 'b') + (() => someBoolean ? 'c' : 'd')() + 'e';
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"restrict-plus-operands": true
	}
}
```

#### Actual behavior

`Operands of '+' operation must either be both strings or both numbers`

#### Expected behavior

No error, as with TS 2.0.10 and tslint 4.1.1.
